app/src/main/java/org/piwigo/helper/URLHelper.java

```diff
@@ -1,41 +1,19 @@
 package org.piwigo.helper;
 
-import android.os.AsyncTask;
-import android.util.Log;
+public class URLHelper
+{
 
-import java.io.IOException;
-import java.net.HttpURLConnection;
-import java.net.URL;
-import java.net.UnknownHostException;
+    public static URLHelper INSTANCE;
 
-import javax.net.ssl.HttpsURLConnection;
-import javax.net.ssl.SSLException;
-
-public class URLHelper extends AsyncTask<String, Void, String> {
-
-    private AsyncUrlResponse delegate;
-
-    public URLHelper(AsyncUrlResponse delegate) {
-        this.delegate = delegate;
+    public URLHelper()
+    {
+        INSTANCE = this;
     }
 
-    @Override
-    protected String doInBackground(String... url) {
-        String newUrl = url[0];
-        newUrl = getPiwigoBaseFor(newUrl);
-
-        newUrl = newUrl.replaceAll("https://", "").replaceAll("http://", "");
-        try {
-            if (isHttpsWebsite(newUrl))
-                return ("https://" + newUrl);
-            else if (isHttpWebsite(newUrl))
-                return ("http://" + newUrl);
-            else
-                return ("https://" + newUrl);
-        } catch (IOException e) {
-            Log.e("URLHelper", "IOException", e);
-            return ("https://" + newUrl);
-        }
+    public String getUrlWithMethod(String url, String method)
+    {
+        url = url.replaceAll("https://", "").replaceAll("http", "");
```

Comment:
it looks a bit weird that you remove the "://" also in case of https but not if the url is given with http ...

edit: Also should we catch upper case protocol-specifications like "HTTP://"?

Reply:
Yup my error indeed

And yes, let's put everything to lowercase.